### ISSUE TITLE
`az repos pr show` not including tags/labels

### ISSUE THREAD

**Issue by celluj34** [Azure/azure-cli#23626]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**
`az repos pr show --id 2582 --detect true`

**Describe the bug**
Tags (or, as they are defined in the documentation, Labels) are not being returned in the above command. They are being returned from the REST api at `.../_apis/git/repositories/<repository>/pullRequests/<pr id>/labels`

**To Reproduce**
1. Create a pull request in Azure DevOps
2. Add a tag. The specific text doesn't matter.
3. Run the above command to show the details

**Expected behavior**
I expect the JSON response to contain any and all tags I've added to the PR.

**Environment summary**
n/a

**Additional context**
n/a

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
Adding Service attention label to devops issues.

@v-anvashist, @V-hmusukula. Could you please look into this and provide an update ?

**Comment by v-anvashist** [comment]
@celluj34 We are receiving the response from API so we have raised the issue to API team.

Once will get the update from them will share the updates on the same.

**Comment by v-soujanya** [comment]
@celluj34 we are following up with API team on this issue as we get the response from API. Once we receive the response from API team, we will share the updates.

**Comment by celluj34** [comment]
Great, thank you! Hopefully good news soon 🤞

**Comment by v-soujanya** [comment]
@celluj34, we have not received response from API team yet and we are continuously following up with the team. we will share the updates once we receive from the API team.

**Comment by v-soujanya** [comment]
@celluj34
We have received response from API team and currently they have no plans enhancing the CLI to report the tags associated to the PR. They will consider this request if they have bandwidth.
Thanks!

**Comment by celluj34** [comment]
Well, that's disappointing. Would the priority change if it were considered a bug instead of a feature? My reasoning would be since the call returns other information about the pull request (including nested details on the repository), _not_ returning the tags is a "incomplete" response.

**Comment by v-anvashist** [comment]
@celluj34 We will discuss with API team and get back to you.

**Comment by Arvindgade** [comment]
@celluj34 do we have any work item raised for this?

**Comment by celluj34** [comment]
@Arvindgade I am not sure what you mean - I don't work for Microsoft, I just use the cli.

**Comment by celluj34** [comment]
> @celluj34 We will discuss with API team and get back to you.

Has there been discussion on this?I'm still interested in seeing this added.

**Comment by celluj34** [comment]
> @celluj34 We will discuss with API team and get back to you.

@v-anvashist Has there been discussion on this? I'm still interested in seeing this added.

**Comment by killergege** [comment]
I agree that this should be considered a bug. The field "labels" is there, so we are expecting this field to be properly initialized, but always empty.

**Comment by v-anvashist** [comment]
Hi @celluj34 We received response from API Team 
<img width="476" alt="image" src="https://github.com/Azure/azure-cli/assets/98596197/8d3eddad-2b37-46f7-8819-8fc6156a9b1a">

**Comment by hermanho** [comment]
It is interesting. From the API spec, there has a field "labels" but no value in the response.
https://learn.microsoft.com/en-us/rest/api/azure/devops/git/pull-requests/get-pull-request-by-id?view=azure-devops-rest-7.1&tabs=HTTP#gitpullrequest

Even though the labels cannot be query in get-pull-request-by-id, the labels can be retrieved from the pull request labels API and aggregate it together in cli level. Why not go to this way?

**Comment by celluj34** [comment]
@hermanho Do you mean, why can't I combine the results from two queries? That's what I'm doing now, but I shouldn't have to. Using the az cli from Azure DevOps is much easier than hitting the rest api. And, as you've found, there is already the labels object but it's not populated. It should be returning the existing information.

Honestly, I don't understand why something like this requires years of deliberation. The rest api is returning this information, the cli has a placeholder for this information. Just... make them equal parity.

**Comment by ducktordanny** [comment]
It's a bit funny that this is considered a feature. We get an incorrect value (`null`) for an existing property. :D

**Comment by celluj34** [comment]
@ducktordanny nice, thanks for the cli command! Here's the full command I got working:

```
az repos pr list -s <feature branch> -t master --top 1 --query "[].labels[?active].name | [0]"
```

returns you an output like so:

```json
[
  "tag1",
  "tag2"
]
```

**Comment by ducktordanny** [comment]
Cool, I've also written a post on Medium which is quite related, I'll leave it here, in case it can be a help to anyone: [Stuck Azure pipeline cache](https://medium.com/@ducktordanny/stuck-azure-pipeline-cache-86770d3723a0).

**Comment by thomas11** [comment]
This is really a bug, not a feature, and should be fixed. You're returning incorrect data.
